src/app/settings/views/Users/UserForm/UserForm.test.tsx

```diff
@@ -135,25 +158,45 @@ describe("UserForm", () => {
         </MemoryRouter>
       </Provider>
     );
-    wrapper.find(BaseUserForm).invoke("onSave")({
-      isSuperuser: true,
-      email: "test@example.com",
-      fullName: "Miss Wallaby",
-      password: "test1234",
-      passwordConfirm: "test1234",
-      username: "admin",
-    });
 
+    await userEvent.click(
+      screen.getByRole("button", { name: "Change password…" })
+    );
+
+    await userEvent.type(screen.getByText("Password"), "test1234");
+
+    await userEvent.type(screen.getByText("Password (again)"), "test1234");
+
+    await userEvent.click(screen.getByRole("button", { name: "Save user" }));
+
+    // TODO: Fix form submitting "user/update" action twice:
+    // https://github.com/canonical/app-tribe/issues/1254
```

Comment:
It looks like this is happening when you do:
```
await userEvent.click(
      screen.getByRole("button", { name: "Change password…" })
    );
```

Which results in this warning:

```
      Warning: You submitted a Formik form using a button with an unspecified `type` attribute.  Most browsers default button elements to `type="submit"`. If this is not a submit button, please add `type="button"`.
```

This doesn't appear to submit the form when in the browser for me (at least when I tried in Chrome), but it fixes the test if I add `type="button"` here: https://github.com/canonical/maas-ui/blob/main/src/app/base/components/UserForm/UserForm.tsx#L170

Reply:
Ohhhh I was wondering why that was happening!